### PR TITLE
fixes #1310 and improves upon ideas shared on #1450

### LINKED ISSUE #1310: compiled app's views aren't executed
This PR https://github.com/web2py/web2py/pull/1194 broke compiled apps. 
Before reverting and releasing, I propose a reassessment of the whole compileapp.py (seems that we can order both code and "thoughs" once and for all). I opened a relevant discussion on web2py-developers

### PR DESCRIPTION
**Pull request by niphlod**

### CODE REVIEW

**Comment by niphlod** [comment]
@ilvalle : since you have been the man in charge of #1450 can you take a look at this ?

**Comment by ilvalle** [comment]
Can't try right now but it looks ok. Btw don't know what is _TEST function.

**Comment by ilvalle** [comment]
To have the same behavior there is in gae, I'd add a call to getcfs at line 722.

**Comment by niphlod** [comment]
hem, do we WANT the same behaviour as in GAE ?

**Comment by mdipierro** [comment]
The test function allows to run doctests from admin. Wan't this break backward compatibility for compiled apps? I am on with it but we should make a very prominent note of it.

**Comment by niphlod** [comment]
if it's not final, wouldn't have been better to continue the discussion before merging ?

**Comment by mdipierro** [comment]
I will revert if needed. We can still discuss.

**Comment by niphlod** [comment]
NP: I'll open a new issue to keep the track or this will get lost, given the "merged" status is normally hidden

**Comment by ilvalle** [comment]
I meant something like
```
diff --git a/gluon/compileapp.py b/gluon/compileapp.py
index 42794ba..cbd584a 100644
--- a/gluon/compileapp.py
+++ b/gluon/compileapp.py
@@ -724,9 +724,10 @@ def run_view_in(environment):
                                             pjoin(folder, 'views'),
                                             context=environment), layer))
         else:
-            ccode = parse_template(view,
-                                   pjoin(folder, 'views'),
-                                   context=environment)
+            ccode = getcfs(view, view,
+                           lambda: compile2(parse_template(view,
+                                            pjoin(folder, 'views'),
+                                            context=environment), view))
         restricted(ccode, environment, layer)
```

**Comment by niphlod** [comment]
@ilvalle : I went for including cfs directly in template.py . should result in the same thing without overcharging with lambda statements ^_^ . or not ?

**Comment by ilvalle** [comment]
I see the proposed one more straightforward because all files (models, controllers, views), plain_text/compiled, are first cached and then executed.

**Comment by niphlod** [comment]
@ilvalle: I see. Would you make a PR reverting changes to template and adding your own to compileapp ?
I'm tackling autocompletewidget....

File: gluon/compileapp.py [hunk 1]
```diff
@@ -691,18 +686,18 @@ def run_view_in(environment):
     else:
         filename = pjoin(folder, 'views', view)
         if os.path.exists(path): # compiled views
-            x = view.replace('/', '_')
-            files = ['views_%s.pyc' % x]
+            x = view.replace('/', '.')
+            files = ['views.%s.pyc' % x]
```
Comment (ilvalle) [inline gluon/compileapp.py]: compiled views use ```_``` as separator. for example ```views_default_index.html```
https://github.com/web2py/web2py/blob/master/gluon/compileapp.py#L490

**Pull request merged**